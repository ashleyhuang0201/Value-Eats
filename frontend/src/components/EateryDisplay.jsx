import React from "react";
import {
  Card,
  Grid,
  CardContent,
  CardHeader,
  CardMedia,
  Typography,
  makeStyles,
} from "@material-ui/core";
import { useHistory } from "react-router-dom";
import RatingWithNum from "../components/RatingWithNum";

const useStyles = makeStyles({
  wideCard: {
    marginTop: "20px",
    transition: "transform 0.15s ease-in-out",
    "&:hover": {
      transform: "scale3d(1.02, 1.02, 1)",
      cursor: "pointer",
    },
  },
});

export default function EateryDisplay ({
  name,
  id,
  discount,
  cuisines,
  rating,
}) {
  const classes = useStyles();
  const history = useHistory();
  return (
        <Card
            className={classes.wideCard}
            onClick={(e) =>
              history.push({
                pathname: `/EateryProfile/${name}/${id}`,
              })
            }
        >
            <CardHeader title={"UP TO " + discount + " OFF"} />
            <CardMedia
                style={{
                  height: "150px",
                }}
                // TODO change display image for restaurant
                image={
                    "https://i.pinimg.com/originals/b8/e1/4a/b8e14a14af9434aa5ccc0376a47a5237.jpg"
                }
            />
            <CardContent>
                <Grid container justify="space-between" alignItems="flex-end">
                    <Grid item xs={8}>
                        <Typography variant="h5">{name}</Typography>
                        <Typography variant="subtitle2">
                            {cuisines.join(", ")}
                        </Typography>
                    </Grid>
                    <Grid item>
                        <RatingWithNum rating={rating} />
                    </Grid>
                </Grid>
            </CardContent>
        </Card>
  );
}
