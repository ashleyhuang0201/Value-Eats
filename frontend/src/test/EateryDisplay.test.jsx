import { shallow } from "enzyme";
import React from "react";
import renderer from "react-test-renderer";
import EateryDisplay from "../components/EateryDisplay";

describe("EateryDisplay", () => {
  it("can work in the base case", () => {
    const display = shallow(<EateryDisplay name={"Test Eatery"}
      id={0}
      discount={"50%"}
      cuisines={["Chinese", "Japanese"]}
      rating={2.5}/>);
    expect(display.exists()).toBeTruthy();
  });

  it("displays the discount correctly", () => {
    const display = shallow(<EateryDisplay name={"Test Eatery"}
      id={0}
      discount={"50%"}
      cuisines={["Chinese", "Japanese"]}
      rating={2.5}/>);
    console.log(display.children().at(0).prop("title"));

    expect(display.children().at(0).prop("title") === "UP TO 50% OFF");
  });

  it("displays the eateries correctly", () => {
    const display = shallow(<EateryDisplay name={"Test Eatery"}
      id={0}
      discount={"50%"}
      cuisines={["Chinese", "Japanese"]}
      rating={2.5}/>);
    expect(display.text().includes("Chinese, Japanese")).toBe(true);
  });

  /* Snapshot tests - creates snapshots for new tests, otherwise will check the existing tests. To update existing tests,
  follow the prompts on the screen after running test. Done to avoid regression. */

  it("renders basic component", () => {
    const display = renderer.create(<EateryDisplay name={"Test Eatery"}
      id={0}
      discount={"50%"}
      cuisines={["Chinese", "Japanese"]}
      rating={2.5}
    />);
    expect(display).toMatchSnapshot();
  });
});
