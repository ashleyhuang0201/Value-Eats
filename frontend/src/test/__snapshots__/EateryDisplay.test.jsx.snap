// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EateryDisplay renders basic component 1`] = `
<div
  className="MuiPaper-root MuiCard-root makeStyles-wideCard-1 MuiPaper-elevation1 MuiPaper-rounded"
  onClick={[Function]}
>
  <div
    className="MuiCardHeader-root"
  >
    <div
      className="MuiCardHeader-content"
    >
      <span
        className="MuiTypography-root MuiCardHeader-title MuiTypography-h5 MuiTypography-displayBlock"
      >
        UP TO 50% OFF
      </span>
    </div>
  </div>
  <div
    className="MuiCardMedia-root"
    style={
      Object {
        "height": "150px",
      }
    }
  />
  <div
    className="MuiCardContent-root"
  >
    <div
      className="MuiGrid-root MuiGrid-container MuiGrid-align-items-xs-flex-end MuiGrid-justify-content-xs-space-between"
    >
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-8"
      >
        <h5
          className="MuiTypography-root MuiTypography-h5"
        >
          Test Eatery
        </h5>
        <h6
          className="MuiTypography-root MuiTypography-subtitle2"
        >
          Chinese, Japanese
        </h6>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item"
      >
        <div
          className="MuiGrid-root MuiGrid-container MuiGrid-item"
        >
          <div
            className="MuiGrid-root MuiGrid-item"
          >
            <span
              aria-label="2.5 Stars"
              className="MuiRating-root MuiRating-readOnly"
              onMouseLeave={[Function]}
              onMouseMove={[Function]}
              role="img"
            >
              <span
                className="MuiRating-decimal"
              >
                <span
                  style={
                    Object {
                      "overflow": "hidden",
                      "position": "absolute",
                      "width": "0%",
                      "zIndex": 1,
                    }
                  }
                >
                  <span
                    className="MuiRating-icon MuiRating-iconFilled"
                  >
                    <svg
                      aria-hidden={true}
                      className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                      />
                    </svg>
                  </span>
                </span>
                <span
                  style={Object {}}
                >
                  <span
                    className="MuiRating-icon MuiRating-iconFilled"
                  >
                    <svg
                      aria-hidden={true}
                      className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                      />
                    </svg>
                  </span>
                </span>
              </span>
              <span
                className="MuiRating-decimal"
              >
                <span
                  style={
                    Object {
                      "overflow": "hidden",
                      "position": "absolute",
                      "width": "0%",
                      "zIndex": 1,
                    }
                  }
                >
                  <span
                    className="MuiRating-icon MuiRating-iconFilled"
                  >
                    <svg
                      aria-hidden={true}
                      className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                      />
                    </svg>
                  </span>
                </span>
                <span
                  style={Object {}}
                >
                  <span
                    className="MuiRating-icon MuiRating-iconFilled"
                  >
                    <svg
                      aria-hidden={true}
                      className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                      />
                    </svg>
                  </span>
                </span>
              </span>
              <span
                className="MuiRating-decimal"
              >
                <span
                  style={
                    Object {
                      "overflow": "hidden",
                      "position": "absolute",
                      "width": "50%",
                      "zIndex": 1,
                    }
                  }
                >
                  <span
                    className="MuiRating-icon MuiRating-iconFilled"
                  >
                    <svg
                      aria-hidden={true}
                      className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                      />
                    </svg>
                  </span>
                </span>
                <span
                  style={Object {}}
                >
                  <span
                    className="MuiRating-icon MuiRating-iconEmpty"
                  >
                    <svg
                      aria-hidden={true}
                      className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
                      focusable="false"
                      style={
                        Object {
                          "color": "#FFB400",
                        }
                      }
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M22 9.24l-7.19-.62L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21 12 17.27 18.18 21l-1.63-7.03L22 9.24zM12 15.4l-3.76 2.27 1-4.28-3.32-2.88 4.38-.38L12 6.1l1.71 4.04 4.38.38-3.32 2.88 1 4.28L12 15.4z"
                      />
                    </svg>
                  </span>
                </span>
              </span>
              <span
                className="MuiRating-decimal"
              >
                <span
                  style={
                    Object {
                      "overflow": "hidden",
                      "position": "absolute",
                      "width": "0%",
                      "zIndex": 1,
                    }
                  }
                >
                  <span
                    className="MuiRating-icon MuiRating-iconEmpty"
                  >
                    <svg
                      aria-hidden={true}
                      className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
                      focusable="false"
                      style={
                        Object {
                          "color": "#FFB400",
                        }
                      }
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M22 9.24l-7.19-.62L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21 12 17.27 18.18 21l-1.63-7.03L22 9.24zM12 15.4l-3.76 2.27 1-4.28-3.32-2.88 4.38-.38L12 6.1l1.71 4.04 4.38.38-3.32 2.88 1 4.28L12 15.4z"
                      />
                    </svg>
                  </span>
                </span>
                <span
                  style={Object {}}
                >
                  <span
                    className="MuiRating-icon MuiRating-iconEmpty"
                  >
                    <svg
                      aria-hidden={true}
                      className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
                      focusable="false"
                      style={
                        Object {
                          "color": "#FFB400",
                        }
                      }
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M22 9.24l-7.19-.62L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21 12 17.27 18.18 21l-1.63-7.03L22 9.24zM12 15.4l-3.76 2.27 1-4.28-3.32-2.88 4.38-.38L12 6.1l1.71 4.04 4.38.38-3.32 2.88 1 4.28L12 15.4z"
                      />
                    </svg>
                  </span>
                </span>
              </span>
              <span
                className="MuiRating-decimal"
              >
                <span
                  style={
                    Object {
                      "overflow": "hidden",
                      "position": "absolute",
                      "width": "0%",
                      "zIndex": 1,
                    }
                  }
                >
                  <span
                    className="MuiRating-icon MuiRating-iconEmpty"
                  >
                    <svg
                      aria-hidden={true}
                      className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
                      focusable="false"
                      style={
                        Object {
                          "color": "#FFB400",
                        }
                      }
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M22 9.24l-7.19-.62L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21 12 17.27 18.18 21l-1.63-7.03L22 9.24zM12 15.4l-3.76 2.27 1-4.28-3.32-2.88 4.38-.38L12 6.1l1.71 4.04 4.38.38-3.32 2.88 1 4.28L12 15.4z"
                      />
                    </svg>
                  </span>
                </span>
                <span
                  style={Object {}}
                >
                  <span
                    className="MuiRating-icon MuiRating-iconEmpty"
                  >
                    <svg
                      aria-hidden={true}
                      className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
                      focusable="false"
                      style={
                        Object {
                          "color": "#FFB400",
                        }
                      }
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M22 9.24l-7.19-.62L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21 12 17.27 18.18 21l-1.63-7.03L22 9.24zM12 15.4l-3.76 2.27 1-4.28-3.32-2.88 4.38-.38L12 6.1l1.71 4.04 4.38.38-3.32 2.88 1 4.28L12 15.4z"
                      />
                    </svg>
                  </span>
                </span>
              </span>
            </span>
          </div>
          <div
            className="MuiGrid-root MuiGrid-item"
            style={
              Object {
                "marginLeft": "10px",
                "marginTop": "4px",
              }
            }
          >
            2.5
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
